packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -18,14 +18,16 @@ import { ITheme, createTheme } from '../../Styling';
 
 const getClassNames = classNamesFunction<IFabricStyleProps, IFabricStyles>();
 const getFabricTheme = memoizeFunction((theme?: ITheme, isRTL?: boolean) => createTheme({ ...theme, rtl: isRTL }));
-const getDir = memoizeFunction((theme?: ITheme, dir?: IFabricProps['dir']) => {
-  if (dir) {
-    return dir;
-  }
-  if (theme && theme.rtl !== undefined) {
-    return theme.rtl ? 'rtl' : 'ltr';
-  }
-  return getRTL() ? 'rtl' : 'ltr';
+
+const getAllDir = memoizeFunction((theme?: ITheme, dir?: IFabricProps['dir']) => {
```

Comment:
why are we memoizing the function?

Reply:
it was memoized when i got there :D

Reply:
Original call was memoized, so I copied that pattern.

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx#L20

Reply:
but i can see how a large theme is quite different than this call..will remove